### PR TITLE
Setup

### PR DESCRIPTION
**Pull request by Yorsyboy**
### Description
In this PR i created a Hello-rails-react-app:

**General requirements**
- [x]  There are no linter errors.
- [x]  Followed the Github flow.
- [x]  The documentation is professional.

**Project requirements**

- [x] Create a new Rails app called 'hello-rails-react'.
- [x] Initialize your project with Git.
- [x] Set up your project with webpacker the react-rails gems
- [x] Make sure that your project has a Postgres database set up.
- [x] Create a table for storing your messages and create 5 different greetings.
- [x] Create an API endpoint that selects a random greeting from your table (you will need a controller with an action and Rails route).
- [x] Create a static view that will be the root of your app.
- [x] Create the App component with react-router and render it in your static view.
- [x] Create the Greeting component that will display a greeting. Set it up as a route in your App component.
- [x] Create a store, an action, and a reducer that will connect to your API endpoint to get the random greeting.
- [x] Display the random greeting in your Greeting component.

### CODE REVIEW

**Comment by ahmetbozaci** [comment]
# ❌ ❌ ❌ STATUS: INVALID  ❌ ❌ ❌ 


### Hello @Yorsyboy  ✋🏼 
#### You have submitted a project, that does not include a pull request description. Kindly add it and resubmit. 

#### Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏


### Cheers and Happy coding! 💻 👏
coding


___

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification._

_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._

_Invalid Code Review Request **does not count** into the code reviews limit._

___

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form.](https://airtable.com/shrQAqnBwek5a0O0s)_

[GitHub](https://github.com/ahmetbozaci) | [LinkedIn](https://linkedin.com/in/ahmetbozaci) | [Twitter](https://twitter.com/ahmtbozaci)

_Reviewed by [Ahmet Bozacı ](https://github.com/ahmetbozaci)_

File: db/schema.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  # These are extensions that must be enabled in order to support this database
+  enable_extension "plpgsql"
+
+  create_table "greetings", force: :cascade do |t|
+    t.string "message"
+    t.datetime "created_at", null: false
+    t.datetime "updated_at", null: false
+  end
+
+end
```
Comment (BenMukebo) [inline db/schema.rb]: - [ ]  Great job so far 👏🏻 however a database table for storing different greetings should be created. Actually, I don't see any migration file for a table and there is not any model created. Kindly let's add these 👍🏻

File: config/routes.rb [hunk 2]
```diff
@@ -1,6 +1,10 @@
 Rails.application.routes.draw do
-  # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
+  namespace :api, defaults: { format: 'json' } do
+    resources :greetings, only: %i[index]
+  end
```
Comment (BenMukebo) [inline config/routes.rb:4]: - [ ] Let's create a main static route. (should be marked as `root`). The root webpage is the one that displays requests that include only the domain/port part of the URL and not the resource path (e.g., http://localhost:3000/). Kindly add this 😉 👍🏻 

- [ ] And you need to make sure that a static controller has one action !

File: app/views/static/index.html.erb [hunk 3]
```diff
@@ -0,0 +1,2 @@
+<%= javascript_pack_tag 'application' %>
+<%= react_component('App') %>
```
Comment (BenMukebo) [inline app/views/static/index.html.erb]: - [ ] The view for the root route is created (`root#index`) however It should include only `<div id="root">` The rest should be done with React and Redux.

File: app/controllers/static_controller.rb [hunk 4]
```diff
@@ -0,0 +1,2 @@
+class StaticController < ApplicationController
+end
```
Comment (BenMukebo) [inline app/controllers/static_controller.rb]: - [ ] You need to create an index action here for your root route (`root#index`)

**Review by BenMukebo — CHANGES_REQUESTED** [review]
### Hi @Yorsyboy,

You did a great job so far 👏🏼 👏🏼
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

### To highlight 💪🏻

- Your Pull Request has a proper title and description. ✔️  
- No Linter errors :heavy_check_mark: 
- There is a `Greeting` component. ✔️
- The `App`component has a route that leads to the `Greeting` component.
 👍  Good job!


## STATUS: CHANGES REQUESTED  ♻️ 
_Please Check the comments under the review and make all required changes_

- [ ] Please kindly update your `readme file` to reflect the changes for this milestone as this will make your `readme file` more informative and professional. Check out this [template](https://github.com/microverseinc/readme-template) and adapt it to your project. Replace all the information in the various sections of the template with information pertaining to your project. 

- [ ]  Please, add a `project Title` and a meaningful  `Project description`. Remember A project description is a high-level overview of why you're doing a project. The document explains a project's objectives and essential qualities. Is an essential document that provides your audience with a comprehensive overview of the project. Think of it as the elevator pitch that focuses on what and why without delving into how. 😉 💯 
- [ ] It will be nice to add explicit steps to clone and run your project locally, as this will be of help to non-technical users. You can put the steps to clone and run the project in a new Getting Started section under the project's description. This includes
- `git clone URL` , `cd project_name`,
- `bundle install`
- `rails db:create db:migrate db:seed`
- `yarn install`
- `...`

etc 🙃  Please refer to this [template](https://github.com/microverseinc/readme-template), you can copy and paste the content but remember to customize it to your project.

- The README file It is the first file a person will see when they encounter your project, so it should be fairly brief but detailed. It will make your project stand out from a bunch of others.


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._




Cheers and Happy coding!👏👏👏 💻  
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please ping me @BenMukebo when you comment so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Davidosky007 — APPROVED** [review]
Hi @Yorsyboy 👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
* No linter errors 👍 
* Well written PR description 👍 
* Proper use of Gitflow 👍 
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**